docs/deployment.md

```diff
@@ -0,0 +1,483 @@
+KubeVirt Deployment
+=
+
+# Preface
+When deploying [KubeVirt](https://kubevirt.io/) onto a [Kubernetes](https://kubernetes.io/) cluster, we deploy the following:
+
+- `virt-operator` and its dependencies
+- KubeVirt CustomResource
+
+This is done using the [kubectl](https://kubernetes.io/docs/reference/kubectl/) command line tool or programmatically.
+
+```shell
+# Point at latest stable release
+export RELEASE=$(curl https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)
+# Deploy the virt-operator
+kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-operator.yaml
+# Create the KubeVirt CR (instance deployment request) which triggers the actual installation
+kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-cr.yaml
+# Wait until all KubeVirt components are up
+kubectl --namespace=kubevirt wait kubevirt kubevirt --for condition=Available
+```
+
+# Virt Operator Deployment
+
+`virt-operator` is the component responsible for deploying the components of the KubeVirt project.
+
+## Namespace
+A dedicated [Namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) is created for KubeVirt's components.
+
+```shell
+kubectl get namespace kubevirt -o yaml
+```
+
+## KubeVirt CustomResourceDefinition
+KubeVirt extends Kubernetes using [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (CRDs).
+
+The first deployed CRD is `kubevirts.kubevirt.io` which allows us to later create the `KubeVirt` CustomResource.
+
+```shell
+kubectl get customresourcedefinitions.apiextensions.k8s.io kubevirts.kubevirt.io -o yaml
+```
+
+## Priority Class
+`kubevirt-cluster-critical` [PriorityClass](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) is deployed and will be used by `virt-operator`.
+
+```shell
+kubectl get priorityclasses.scheduling.k8s.io kubevirt-cluster-critical -o yaml
+```
+
+## ServiceAccount
+A [ServiceAccount](https://kubernetes.io/docs/concepts/security/service-accounts/) is deployed to be used by `virt-operator`.
+
+```shell
+kubectl get --namespace=kubevirt serviceaccount kubevirt-operator -o yaml
+```
+
+## ClusterRoles
+
+```shell
+kubectl get clusterroles.rbac.authorization.k8s.io kubevirt.io:operator -o yaml
+kubectl get clusterroles.rbac.authorization.k8s.io kubevirt-operator -o yaml
+```
+
+## ClusterRoleBinding
+
+Binds the `kubevirt-operator` ClusterRole to the `kubevirt:kubevirt-operator` ServiceAccount.
+
+```shell
+kubectl get clusterrolebindings.rbac.authorization.k8s.io kubevirt-operator -o yaml
+```
+
+## Role
+
+```shell
+kubectl get --namespace=kubevirt roles.rbac.authorization.k8s.io kubevirt-operator -o yaml
```

Comment:
I'd also use singular instead of plural when getting a specific object and, like `-o`, I feel that `-n` is widespread enough to avoid using the long version.
```suggestion
kubectl get role kubevirt-operator -n kubevirt -o yaml
```
That reads almost like normal text, i.e. "get the role named kubevirt-operator in the kubevirt namespace".

I realize this is very nit-picky and I wont insist on any of it, it's really just personal preference.

Reply:
Done.